syntax/grammar

```diff
@@ -0,0 +1,52 @@
+
+use_statement: identifier ['as' identifier]
+
+function_def: 'fn' NAME '(' decl_list ')' type NEWLINE function_body
+
+return_statement: 'ret' [expression]
+
+decl_list: decl | decl "," decl_list | epsilon
+
+type: i64 | i32 | i16 | i8 | i1 | f32 | f64 | str | '[' type ']' | void | NAME
```

Comment:
Yep, looking at it again I'm not sure why I was questioning it. That's accurate.

Dictionaries are not part of the language itself - at least, not yet. I'd love an RFC about them. My hesitations lay with creating a syntactical construct that is tightly tied to something from the standard library.